### ISSUE TITLE
Hash for filename differs on 32-bit and 64-bit devices

### ISSUE THREAD

**Issue by brockboland** [vokal/VOKMockUrlProtocol#28]
I haven't done any digging into this, but have confirmed that tests passing on a iPhone 4S or 5 fail on 5S and above, because the hash generated for the mock data filenames is different.

**Comment by vokal-isaac** [comment]
Do you have example(s) of the requests that lead to different hashes?

**Comment by brockboland** [comment]
Ohhhhhhh nevermind.

iPhone 5, iOS 8.4:
For URL http://127.0.0.1/testing/app/security/usersessions/membersessions:
Body: {"beneficiaryLastName":"sample","beneficiaryFirstName":"jane","birthDate":"1980-02-01T12:00:00.000Z"}
Hash: 60a33a0e11a7222b9406a7536b40bd38f76254663b91f80868b6ca5338b9e9dd

iPhone 5S, iOS 8.4:
For URL http://127.0.0.1/testing/app/security/usersessions/membersessions:
Body: {"beneficiaryFirstName":"jane","beneficiaryLastName":"sample","birthDate":"1980-02-01T12:00:00.000Z"}
Hash: d4371fcb109ef3773780e29c3ba79e0d28864d5d09525401158934f8f53f5f75

I missed that earlier: [Mantle](https://github.com/Mantle/Mantle) doesn't sort keys the same when generating the JSON for the model object. My bad.

**Comment by vokal-isaac** [comment]
Based on @chillpop's comment here https://github.com/vokal/MaximusEnrollment-iOS/pull/20#issuecomment-124585627, I think it's kind of a bug and that we probably should, when the body is JSON, hash the stable bencoded string rather than hashing the raw body... but I'm not entirely sure.

**Comment by vokal-isaac** [comment]
@brockboland Is the request being sent with content type `application/json`?

**Comment by brockboland** [comment]
@vokal-isaac yes. Are you saying that VOKMockURLProtocol should (or does?) convert the JSON back to an array or dictionary, and hash that instead?

**Comment by vokal-isaac** [comment]
When the body is JSON (content type `application/json`), the mock URL protocol deserializes the JSON and bencodes that data structure, specifically because the order of items in an `NSDictionary` isn't stable across platforms, to construct the primary file name (the sort-of readable one).  At present, the alternate hash filename is based on hashing the raw body.  I'm suggesting that in the JSON case, we should instead hash the bencoded string value of the body.

**Comment by brockboland** [comment]
I think I'm misunderstanding the last bit of this. Wouldn't we want to hash the bencoded data structure, not the body string? Either way, this should probably be a new issue, and you obviously get this better than I do: could you open one?

**Comment by vokal-isaac** [comment]
Yeah, that's what I meant—we should be hashing the bencoded version of the body (at least, when the body is JSON), but instead we're hashing the body itself.  I think this issue captures what's going on and is definitely enough to remind me what needs to be fixed.
